main.tf

```
@@ -38,6 +38,34 @@ data "aws_iam_policy_document" "assume_role" {
   }
 }
 
+data "aws_iam_policy_document" "amazon_eks_worker_node_autoscaler_policy" {
+  count = (var.enabled && var.enable_cluster_autoscaler) ? 1 : 0
+  statement {
+    sid = "AllowToScaleEKSNodeGroupAutoScalingGroup"
+
+    actions = [
+      "autoscaling:DescribeAutoScalingGroups",
+      "autoscaling:DescribeAutoScalingInstances",
+      "autoscaling:DescribeLaunchConfigurations",
+      "autoscaling:DescribeTags",
+      "autoscaling:SetDesiredCapacity",
+      "autoscaling:TerminateInstanceInAutoScalingGroup",
+      "ec2:DescribeLaunchTemplateVersions"
+    ]
+
+    resources = [
+      "*"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "amazon_eks_worker_node_autoscaler_policy" {
+  count  = (var.enabled && var.enable_cluster_autoscaler) ? 1 : 0
+  name   = "AmazonEKSNodeGroupAutoscaler"
```

```suggestion
  name   = "${module.label.id}-autoscaler"
```

let's use the label module to generate (unique and consistent) names for all resources.

---

please add the new policy attachment (`resource "aws_iam_role_policy_attachment" "amazon_eks_worker_node_autoscaler_policy"`) to the list here
https://github.com/cloudposse/terraform-aws-eks-node-group/blob/master/main.tf#L103